### PR TITLE
Remove the validations from queries

### LINKED ISSUE #382: Remove Validations from Yield Curve Queries
In Utils/Queries#locked-in and Utils/Queries#current the YieldCurveNotFound validation currently exists inside the methods LoadLockedInYieldCurveAsync and LoadCurrentYieldCurveAsync.

The conditions that trigger these validations should be reviewed and the validations should be extracted from the Queries.

Documentation could also be subsequently added to the Locked-In and Current sections for the yield curve.

(FYI: @dcolleoni)

### PR DESCRIPTION
**Pull request by andrey-katz-systemorph**
Removes the validations from the YC queries and adjusts the import storage and tests accordingly

### CODE REVIEW

File: ifrs17/Import/ImportStorage.ipynb [hunk 1]
```diff
@@ -477,6 +479,15 @@
                 "\n    public IEnumerable<string> GetClaims() => hierarchyCache.Get<AmountType>(AmountTypes.CL).Descendants().Select(x => x.SystemName);",
                 "\n    public IEnumerable<string> GetCdr() => hierarchyCache.Get<AmountType>(AmountTypes.CDR).Descendants(includeSelf: true).Select(x => x.SystemName);",
                 "\n    public IEnumerable<string> GetCoverageUnits() => hierarchyCache.Get<AmountType>(AmountTypes.CU).Descendants(includeSelf: true).Select(x => x.SystemName);",
+                "\n",
+                "\n    private YieldCurve LogNoYcError(string dn, bool locked) =>",
+                "\n        DataNodeDataBySystemName.TryGetValue(dn, out var dataNodeData) ?  ",
```
Comment (dcolleoni) [inline ifrs17/Import/ImportStorage.ipynb]: You don't need to validate DataNodeDataBySystemName.TryGetValue(dn:
if you get to here, the DataNode has alredy been validated (parser).

You can use a direct access to the Dictionary

File: ifrs17/Import/ImportStorage.ipynb [hunk 2]
```diff
@@ -477,6 +479,15 @@
                 "\n    public IEnumerable<string> GetClaims() => hierarchyCache.Get<AmountType>(AmountTypes.CL).Descendants().Select(x => x.SystemName);",
                 "\n    public IEnumerable<string> GetCdr() => hierarchyCache.Get<AmountType>(AmountTypes.CDR).Descendants(includeSelf: true).Select(x => x.SystemName);",
                 "\n    public IEnumerable<string> GetCoverageUnits() => hierarchyCache.Get<AmountType>(AmountTypes.CU).Descendants(includeSelf: true).Select(x => x.SystemName);",
+                "\n",
+                "\n    private YieldCurve LogNoYcError(string dn, bool locked) =>",
```
Comment (dcolleoni) [inline ifrs17/Import/ImportStorage.ipynb]: Remove the LogNoYcError method. 
- it is used only twice, 
- taking out the DataNodeValidation(see next comment) is pretty simple
- the bool makes it not nice to read.

File: ifrs17/Import/ImportStorage.ipynb (hunk 1 above)
Comment (andrey-katz-systemorph) [reply ifrs17/Import/ImportStorage.ipynb]: done

File: ifrs17/Import/ImportStorage.ipynb (hunk 2 above)
Comment (andrey-katz-systemorph) [reply ifrs17/Import/ImportStorage.ipynb]: done

File: ifrs17/Import/ImportStorage.ipynb [hunk 3]
```diff
@@ -329,12 +330,17 @@
...
-                "\n            (EconomicBases.L, _ ) => LockedInYieldCurve[id.DataNode],",
+                "\n                                     ? GetCurrentYieldCurve(id.DataNode, CurrentPeriod) ",
+                "\n                                     : GetCurrentYieldCurve(id.DataNode, PreviousPeriod),",
+                "\n            (EconomicBases.C, PeriodType.EndOfPeriod) => GetCurrentYieldCurve(id.DataNode, CurrentPeriod),            ",
+                "\n            (EconomicBases.L, _ ) => LockedInYieldCurve.TryGetValue(id.DataNode, out var yc) && yc != null ? yc : ",
+                "\n                                    (YieldCurve)ApplicationMessage.Log(Error.YieldCurveNotFound, id.DataNode, DataNodeDataBySystemName[id.DataNode].ContractualCurrency, ",
+                "\n                                                                DataNodeDataBySystemName[id.DataNode].Year.ToString(), ",
+                "\n                                                                args.Year == DataNodeDataBySystemName[id.DataNode].Year ? args.Month.ToString() : MonthInAYear.ToString(), ",
```
Comment (dcolleoni) [inline ifrs17/Import/ImportStorage.ipynb]: the month in the error message should always be args.Month.ToString()

**Review by dcolleoni — APPROVED** [review]

File: ifrs17/Import/ImportStorage.ipynb (hunk 3 above)
Comment (andrey-katz-systemorph) [reply ifrs17/Import/ImportStorage.ipynb]: fixed

**Pull request merged**